Rust/src/pointstore.rs

```diff
@@ -1,14 +1,28 @@
-use crate::rcf::Max;
-use std::fmt::Debug;
 extern crate num;
-use crate::intervalstoremanager::IntervalStoreManager;
+use std::{collections::HashMap, convert::TryFrom, fmt::Debug};
 
-use std::collections::HashMap;
+use crate::{intervalstoremanager::IntervalStoreManager, types::Location};
 
-use std::convert::TryFrom;
+pub trait PointStore {
+    fn get_shingled_point(&self, point: &[f32]) -> Vec<f32>;
+    fn get_size(&self) -> usize;
+    fn get_missing_values(&self, values: &[usize]) -> Vec<usize>;
+    fn get_copy(&self, index: usize) -> Vec<f32>;
+    fn is_equal(&self, point: &[f32], index: usize) -> bool;
+    fn get_reference_and_offset(&self, index: usize) -> (&[f32], usize);
+
+    fn add(&mut self, point: &[f32]) -> usize;
+    fn inc(&mut self, index: usize);
+    fn dec(&mut self, index: usize);
+    fn adjust_count(&mut self, result: &[(usize, usize)]);
+    fn compact(&mut self);
+}
 
 #[repr(C)]
-pub struct PointStore<L> {
+pub struct VectorPointStore<L>
```

Comment:
This is correct -- we are using a vector as a pointstore. But there may be a confusion if some one expects that we are storing the points as individual vectors... (which we are not, with shinglesize>1)